modules/apps/object/object-rest-test/src/testFunctional/tests/NestedFieldsDepth.testcase

```diff
@@ -0,0 +1,240 @@
+@component-name = "portal-headless"
+definition {
+
+	property custom.properties = "feature.flag.LPS-165819=true";
+	property portal.release = "true";
+	property portal.upstream = "true";
+	property testray.component.names = "Object";
+	property testray.main.component.name = "Headless";
+
+	setUp {
+		TestCase.setUpPortalInstanceNoSelenium();
+
+		task ("Given Author, Book, Student, Subject and Univeristy object definitions with field 'name' created") {
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "students",
+				name = "Student");
+
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "authors",
+				name = "Author");
+
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "books",
+				name = "Book");
+
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "subjects",
+				name = "Subject");
+
+			ObjectDefinitionAPI.createAndPublishObjectDefinition(
+				en_US_plural_label = "universities",
+				name = "University");
+		}
+
+		task ("And Given studentUsers, studentsUniversities, usersUniversities, studentsSubjects, universityUsers,userStudents, usersSubjects, subjectBooks and booksAuthors relationships created") {
```

Comment:
Just a small concern that the relationships in the "And Given" are not exactly same with the ones we use below, maybe we can update "And Given" to include universitySubjects?

Reply:
The same as Ashley, I think we only write the relationship that we created is ok

Reply:
yes, thank you guys, I forgot to update this step! 🙏🏻 will modify now :)